### PR TITLE
docs: add Contributing

### LINKED ISSUE #696: Add a CONTRIBUTING guidelines
**Do you want to request a *feature* or report a *bug*?**

Documentation.

**What is the expected behavior?**

It would be nice to have a clear `CONTRIBUTING.md` files to follow in order to run the project. It would help external contributors to quickly setup the project. We can take the example of the one we have on [InstantSearch](https://github.com/algolia/instantsearch.js/blob/develop/CONTRIBUTING.md) or simpler one [React InstantSearch](https://github.com/algolia/react-instantsearch/blob/master/CONTRIBUTING.md).

### PR DESCRIPTION
**Pull request by Haroenv**
fixes #696

I based this on the React InstantSearch Contributing and added common pitfalls I could remember off the top of my head.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+yarn test
+```
+
+For integration tests you'll need Algolia credentials as environment variables:
+
+```sh
+ALGOLIA_APP_ID=XXX ALGOLIA_API_KEY=XXX PLACES_APPID=XXX PLACES_APIKEY=XXX node test/run-integration.js
+```
+
+You can also run the tests in your browser, rather than in node (**note that this is currently not working**):
```
Comment (vvo) [inline CONTRIBUTING.md]: Do you mean the tests are not passing?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+```sh
+yarn dev
+```
+
+or
+
+```sh
+yarn dev-integration
```
Comment (vvo) [inline CONTRIBUTING.md]: You also need the env variables here to be setup right?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+```sh
+yarn lint
+```
+
+Files are __not__ automatically formatted with prettier.
+
+## Release
+
+Releasing is done by following the "release API clients" for now; but a contribution that extracts this into a script would be really nice.
```
Comment (vvo) [inline CONTRIBUTING.md]: the ... internal document

File: CONTRIBUTING.md (hunk 1 above)
Comment (Haroenv) [reply CONTRIBUTING.md]: Yes, better like this?

```diff
- **note that this is currently not working**
+ **note that these tests currently not passing**
```

**Comment by Haroenv** [comment]
made your suggested changes @vvo

**Pull request merged**